File: BubbleSort/Go/bubblesort.go
package main

import "fmt"

func main() {
	var temp []int = []int{1, 2, 5, 10, 3, 4, 6}
	BubbleSort(temp)
	fmt.Println(temp)
}

func BubbleSort(numbers []int) {
	swapped := true
	for swapped {
		swapped = false
		for i := 0; i <= len(numbers)-2; i++ {
			if numbers[i] > numbers[i+1] {
				Swap(numbers, i)
				swapped = true
			}
		}

	}
}

func Swap(slc []int, idx int) {
	var tmp = slc[idx]
	slc[idx] = slc[idx+1]
	slc[idx+1] = tmp
}

File: Bubble Sort/Python/bubblesort.py
def bubble_sort(nums):
    nums_copy = nums.copy()
    swapped = True
    end = len(nums_copy)
    while swapped:
        swapped = False
        print(nums_copy)
        for i in range(1, end):
            if nums_copy[i] < nums_copy[i-1]:
                swapped = True
                tmp = nums_copy[i]
                nums_copy[i] = nums_copy[i-1]
                nums_copy[i-1] = tmp
                print(f"Swapped nums_copy[{i}] = {nums_copy[i]} with nums_copy[{i-1}] = {nums_copy[i-1]}")
        end -= 1 
                
    return nums_copy
